### PR TITLE
Prepare Site for Launch

### PR DESCRIPTION
**Pull request by rauljordan**

### CODE REVIEW

File: src/app/web3/web3.service.ts [hunk 1]
```diff
@@ -18,7 +19,15 @@ export const fromWei = w.utils.fromWei;
 export const toBN = w.utils.toBN;
 
 export abstract class Web3Service {
-  constructor(public readonly web3: Web3) {}
+  public depositContractAddress: string;
+
+  constructor(public readonly web3: Web3, private http: HttpClient) {
+    // Initialize the deposit contract address by resolving the
+    // latest from the http endpoint
+    this.http.get(DEPOSIT_CONTRACT_ENDPOINT).subscribe((res: any) => {
```
Comment (prestonvanloon) [inline src/app/web3/web3.service.ts]: Please don't use `any`, this is string, right?

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <nav>
   <mat-toolbar>
-    <span class="brand">Prylabs</span>
+    <span class="brand">Prysmatic Labs Testnet</span>
```
Comment (prestonvanloon) [inline src/app/app.component.html]: This is going to be too big on mobile. Consider using media queries to serve the larger text for desktop and the smaller one for mobile/tablet

File: src/app/participate/participate.component.ts [hunk 3]
```diff
@@ -30,25 +30,27 @@ export class ParticipateComponent implements OnInit {
...
+    private readonly metamask: MetamaskService,
     private readonly snackbar: MatSnackBar,
     private readonly storage: LocalStorage,
     private readonly progress: ProgressService,
     private readonly faucet: FaucetService,
     private readonly cdr: ChangeDetectorRef,
     private readonly formBuilder: FormBuilder,
-  ) { }
+  ) {
+    this.depositContractAddress = this.web3.depositContractAddress;
```
Comment (prestonvanloon) [inline src/app/participate/participate.component.ts]: `this.web3` is not defined until ngOnInit, so you are going to want to make this initialization within `chooseWeb3Provider`

File: src/app/web3/web3.service.ts [hunk 4]
```diff
@@ -42,7 +51,7 @@ export abstract class Web3Service {
 
   /** Reference to the deposit contract */
   get depositContract() {
-    return new this.web3.eth.Contract(DEPOSIT_CONTRACT_ABI as any, DEPOSIT_CONTRACT_ADDRESS);
+    return new this.web3.eth.Contract(DEPOSIT_CONTRACT_ABI as any, this.depositContractAddress);
```
Comment (prestonvanloon) [inline src/app/web3/web3.service.ts]: I think this is going to fail whenever this.depositContractAddress is not yet defined

File: src/app/app.component.html (hunk 2 above)
Comment (prestonvanloon) [reply src/app/app.component.html]: Or just leave it as it was, it's supposed to be a short little logo, not the title of the site.

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,13 +1,12 @@
 <nav>
   <mat-toolbar>
-    <span class="brand">Prylabs</span>
-
-    <a routerLink="/" class="mat-button">Home</a>
-    <a routerLink="/stats" class="mat-button">Stats</a>
-    <a routerLink="/participate" class="mat-button">Participate</a>
+    <span class="brand">Prysm Eth2 Testnet</span>
```
Comment (prestonvanloon) [inline src/app/app.component.html]: I think this is going to be way too crowded on mobile. Please confirm

File: src/app/app.component.ts [hunk 6]
```diff
@@ -11,23 +11,26 @@ import { environment } from '../environments/environment';
...
   ) {
     this.progress.progress.subscribe(v => this.inProgress = v);
   }
 
-  async ngOnInit() {
-    await this.updateValidatorCount();
-
-    this.web3.depositEvents().on('data', () => this.updateValidatorCount());
+  ngOnInit() {
+    this.contractService.getAddress().subscribe(async (res: any) => {
```
Comment (prestonvanloon) [inline src/app/app.component.ts]: Please do not use `any`. `any` is your enemy and it takes all of the benefits away from typescript. It _must_ be a string if you are assigning it to depositContractAddress. Please set it to string.

File: src/app/participate/participate.component.ts [hunk 7]
```diff
@@ -66,14 +71,19 @@ export class ParticipateComponent implements OnInit {
   updateDepositData(data: string) {
     // setItem must be subscribed with a no-op or it won't fire the observable.
     this.storage.setItem(DEPOSIT_DATA_STORAGE_KEY, data).subscribe(() => {});
+    this.stepper.next();
+  }
+
+  onKeyUp(event: any) {
```
Comment (prestonvanloon) [inline src/app/participate/participate.component.ts]: Do not use any

File: src/app/web3/contract.service.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable({
+  providedIn: 'root',
+})
+export class ContractService {
+  private contractSub: Observable<any>;
+
+  constructor(private http: HttpClient) {}
+
+  /** Resolves the latest deposit contract address from the testnet */
+  getAddress(): Observable<any> {
```
Comment (prestonvanloon) [inline src/app/web3/contract.service.ts]: Do not use any

File: src/environments/environment.prod.ts [hunk 9]
```diff
@@ -2,7 +2,6 @@ export const environment = {
   production: true,
   recaptchaSiteKey: '6Lf78pEUAAAAAH4kEmwj-Q9GAT7ILmBFJGSOKkHq',
   apiEndpoint: 'https://api.prylabs.net',
-  depositContractAddress: '0xDc5D60B738AcD12F7bBf752Ea28B2cC2C06A26C7',
   depositAmount: '320000000000000000', // 0.32 ETH
   bootnodeAddress: '/ip4/35.224.249.2/tcp/30001/p2p/QmQEe7o6hKJdGdSkJRh7WJzS6xrex5f4w2SPR6oWbJNriw',
```
Comment (prestonvanloon) [inline src/environments/environment.prod.ts:6]: You can remove this now.

File: src/environments/environment.prod.ts [hunk 10]
```diff
@@ -2,7 +2,6 @@ export const environment = {
   production: true,
   recaptchaSiteKey: '6Lf78pEUAAAAAH4kEmwj-Q9GAT7ILmBFJGSOKkHq',
   apiEndpoint: 'https://api.prylabs.net',
```
Comment (prestonvanloon) [inline src/environments/environment.prod.ts:4]: You can remove this now. (It was for the `beta.prylabs.net/contract`)

File: src/environments/environment.ts [hunk 11]
```diff
@@ -6,7 +6,6 @@ export const environment = {
   production: false,
   recaptchaSiteKey: '6LdX85EUAAAAAGeujVRsy8KJzRvYoC-InusmiDZr',
   apiEndpoint: 'https://api.prylabs.net',
```
Comment (prestonvanloon) [inline src/environments/environment.ts:8]: Same here

File: src/environments/environment.ts [hunk 12]
```diff
@@ -6,7 +6,6 @@ export const environment = {
...
-  depositContractAddress: '0xDc5D60B738AcD12F7bBf752Ea28B2cC2C06A26C7',
   depositAmount: '320000000000000000', // 0.32 ETH
   bootnodeAddress: '/ip4/35.224.249.2/tcp/30001/p2p/QmQEe7o6hKJdGdSkJRh7WJzS6xrex5f4w2SPR6oWbJNriw',
```
Comment (prestonvanloon) [inline src/environments/environment.ts:10]: Same here

File: src/app/app.component.html (hunk 5 above)
Comment (prestonvanloon) [reply src/app/app.component.html]: image


Mobile is going to be the majority of the traffic. It must be usable.

File: src/app/participate/participate.component.html [hunk 13]
```diff
@@ -118,20 +119,14 @@ <h1>
...
-  --demo-config \
-  --bootstrap-node={{BOOTNODE_ADDRESS}}</pre>
+  --datadir=/data
+    </pre>
 
     <span>Run the next command in another terminal.</span>
     <pre>
 docker run -v /tmp/prysm-data:/data --network="host" \
   gcr.io/prysmaticlabs/prysm/validator:{{DOCKER_TAG}} \
-  --beacon-rpc-provider=127.0.0.1:4000 \
```
Comment (prestonvanloon) [inline src/app/participate/participate.component.html:132]: I think you need this one

File: src/app/app.component.html [hunk 14]
```diff
@@ -1,13 +1,12 @@
...
-    <span class="brand">Prylabs</span>
-
-    <a routerLink="/" class="mat-button">Home</a>
-    <a routerLink="/stats" class="mat-button">Stats</a>
-    <a routerLink="/participate" class="mat-button">Participate</a>
+    <span class="brand">💎 Prysm Eth2 Testnet</span>
 
     <span class="flex-spacer"></span>
-    <span class="small-text" *ngIf="numValidators">{{numValidators}} validators!</span>
+    <a routerLink="/" mat-button>Home<mat-icon>home</mat-icon></a>
```
Comment (prestonvanloon) [inline src/app/app.component.html]: I'm not a fan of both the text and the icon. 

I think you could have the icons on mobile only or we could use a hamburger menu. It is looking very crowded

File: src/app/app.component.html [hunk 15]
```diff
@@ -1,13 +1,12 @@
...
+    <a routerLink="/stats" mat-button>Stats<mat-icon>public</mat-icon></a>
+    <a routerLink="/participate" mat-button>Participate<mat-icon>person_add</mat-icon></a>
+    <a target="_blank" href="https://goerli.etherscan.io/address/{{depositContractAddress}}" *ngIf="numValidators" mat-button>{{numValidators}} validators!<mat-icon>launch</mat-icon></a>
```
Comment (prestonvanloon) [inline src/app/app.component.html]: I don't think this looks very good as a button. It's meant to just be some additional/fun information.

File: src/app/home/home.component.scss [hunk 16]
```diff
@@ -1,21 +1,94 @@
...
     text-align: center;
   }
-
   div {
     flex: 1;
   }
 }
+
+.mat-card-content {
+  padding-left: 20px;
```
Comment (prestonvanloon) [inline src/app/home/home.component.scss]: please use 
```css
padding: 0 20px;
```

This is the shorthand for what you've written with padding-right, padding-left

File: src/app/home/home.component.scss [hunk 17]
```diff
@@ -1,21 +1,94 @@
...
+  padding-right: 20px;
+  hr {
+    margin-top: 30px;
```
Comment (prestonvanloon) [inline src/app/home/home.component.scss]:
```css
margin: 0 30px;
```

File: src/app/home/home.component.scss [hunk 18]
```diff
@@ -1,21 +1,94 @@
...
+    text-decoration: none;
+    color: $pink;
+    font-weight: bold;
+  }
+  .social {
+    @include mat-elevation(3);
+    margin-right: 20px;
+    padding-top: 15px;
+    padding-bottom: 18px;
+    padding-left: 14px;
```
Comment (prestonvanloon) [inline src/app/home/home.component.scss]:
```css
padding: 0 14px;
```

File: src/app/home/home.component.scss [hunk 19]
```diff
@@ -1,21 +1,94 @@
...
+    background-color: $black;
+    color: white;
+    font-weight: bold;
+    img {
+      width: 20%;
+      margin-right: 5px;
+    }
+  }
+  p {
+    margin-top: 20px;
```
Comment (prestonvanloon) [inline src/app/home/home.component.scss]:
```css
margin: 20px 0;
```

File: src/assets/GitHub-Logos/README.txt [hunk 20]
```diff
@@ -0,0 +1,30 @@
+GitHub loves third party developers. We want to help make your project awesome, and often that includes showing off GitHub integration with an image. In order to keep everyone happy (and avoid involving lawyers), we've created this repository and filled it with artwork that you may use in your own applications.
```
Comment (prestonvanloon) [inline src/assets/GitHub-Logos/README.txt]: please remove all of these github logos extra stuff

File: src/environments/environment.ts (hunk 11 above)
Comment (rauljordan) [reply src/environments/environment.ts:8]: I believe the api.prylabs.net is for the faucet though

Comment (prestonvanloon) [reply src/environments/environment.ts:8]: oh yeah

**Review by prestonvanloon — APPROVED** [review]
amazing work!

**Pull request merged**